src/utils/eventLevels.js

```diff
@@ -55,17 +55,14 @@ export function eventLevels(rowSegments, limit = Infinity) {
   return { levels, extra }
 }
 
-export function inRange(e, start, end, accessors) {
-  let eStart = dates.startOf(accessors.start(e), 'day')
-  let eEnd = accessors.end(e)
+export function inRange(e, lowerBound, upperBound, accessors) {
+  const start = accessors.start(e)
+  const end = accessors.end(e)
 
-  let startsBeforeEnd = dates.lte(eStart, end, 'day')
-  // when the event is zero duration we need to handle a bit differently
-  let endsAfterStart = !dates.eq(eStart, eEnd, 'minutes')
```

Comment:
this line was preventing a ton of edge case bugs, i'm hesitant to just remove it without understanding how the new logic is equivalent.

Reply:
also probably fine to do a straight getTime() comparison vs minutes

Reply:
Changed my implementation slightly. upperBound is now exclusive, and in case the event start is equal to upperBound, we check if the event has a zero duration, to ensure that the function always returns true if the range and the event have the exact same bounds.

Could you say more about those edge cases? This function is only called in 3 locations (in Month.js, Agenda.js and TimeGrid.js). Other parts of the code call the `inRange` method of `utils/date`.

Reply:
> Could you say more about those edge cases? 

There are a few tests that cover them, but it mostly comes down to events that span weeks or sit on the leading/trailing border of a week. If you don't account for zero duration events specifically they can get dropped from the week or show up at the start and end. If tests are passing tho, that should indicate the case is covered (should)